kong/tools/http.lua

```diff
@@ -553,5 +553,28 @@ do
   end
 end
 
+do
+  local string_sub = string.sub
+
+  ---
+  -- Ensures that a given path adheres to a valid format
+  -- for usage with PDK set_path, or a lua-resty-http client.
+  --
+  -- The function returns the re-formatted path, in its valid form,
+  -- or returns the original string if nothing was changed.
+  --
+  -- @param path string the path to ensure is valid
+  -- @return string the newly-formatted valid path, or the original path if nothing changed
+  function _M.ensure_valid_path(path)
```

Comment:
In the context of `kong.tools.utils`, `ensure_valid_path` is not an appropriate name, as "path validity" is not a well defined concept.  I would suggest `start_with_slash` or something that is more narrow and precise.

Reply:
It was like this because I was thinking to add a bunch more checks later, and the stencil function would make sense for others to use.

I have removed it for now - I can in next round of features if it's still needed.

Reply:
still here, still not working for more than two slashes.  use `gsub`